user_notification/models.py

```diff
@@ -24,22 +26,67 @@ class JobAlert(models.Model):
     def __str__(self):
         return str(self.user_account_id)
 
-    # Calculate the date of the next alert
     @classmethod
     def calc_check_date(cls, last_check_date, frequency_in_days):
+        # Calculate the date of the next alert
         if frequency_in_days < 1:
             myError = ValueError(
                 'frequency_in_days should be Greater than or equal to 1')
             raise myError
         else:
-            if isinstance(last_check_date, datetime):
-                res = last_check_date + timedelta(days=frequency_in_days)
-            else:
+            if isinstance(last_check_date, str):
                 last_date = datetime.strptime(
                     last_check_date, "%Y-%m-%d").date()
                 res = last_date + timedelta(days=frequency_in_days)
+            else:
+                res = last_check_date + timedelta(days=frequency_in_days)
             return res
 
+    @classmethod
+    def jobs_date_check(cls, jobs, last_check_date):
+        # Excludes old job posts by comparing their creation date with last check date
+        if isinstance(last_check_date, str):
+            last_check_date = datetime.strptime(
+                last_check_date, "%Y-%m-%d").date()
+        if isinstance(jobs, QuerySet):
+            for job in jobs:
+                if job.creation_date.date() <= last_check_date:
+                    jobs = jobs.exclude(id=job.id)
+            return jobs
```

Comment:
who calls "jobs_date_check()" ? is it possible to check there if it's empty/not a QuerySet instead of here? (and if it's invalid just don't call this function)

Reply:
I wrote this function because that is what Danielle told me to do in out last google meeting.. check_new_jobs calls this function, and she told me to separate it so there wont be nested 'if'

Reply:
If there are no suitable jobs, the function that filter the jobs return a string saying there are no suitable jobs